service/src/main.rs

```diff
@@ -101,6 +101,24 @@ enum RandomXMode {
     Light,
 }
 
+/// TLS configuration
+///
+/// Either all fields must be specified or none
+#[derive(Args, Debug, Clone)]
+#[group(required = false)]
+pub struct Tls {
+    /// CA certificate
+    #[arg(long, required = false)]
+    pub ca_cert: PathBuf,
+    #[arg(long, required = false)]
+    pub cert: PathBuf,
+    #[arg(long, required = false)]
+    pub key: PathBuf,
+    /// domain name to verify the certificate of server against
+    #[arg(long, default_value = "localhost")]
+    pub domain: String,
```

Comment:
Is this parameter needed for verification? Shouldn't it default to `address`? Additionally the certificate itself says for which domains / IP addresses it is valid. Here is an example server certificate that I've been using for testing:

```bash
$ openssl x509 -text -noout -in server.crt 
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            0a:bf:68:fc:c6:c2:62:3e:ec:5c:80:6f:85:f9:b4:6e:a7:e9:8e:7f
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = EN, ST = Spacemesh, L = Tel Aviv, O = Spacemesh, CN = spacemesh.io, emailAddress = info@spacemesh.io
        Validity
            Not Before: Oct 13 15:52:41 2023 GMT
            Not After : Dec 12 15:52:41 2023 GMT
        // CN below is the server name / address that the certificate is valid for
        Subject: C = EN, ST = Spacemesh, L = Tel Aviv, O = Server, CN = server.spacemesh.io, emailAddress = info@spacemesh.io
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (4096 bit)
                Modulus:
                    // abbreviated
        X509v3 extensions:
            X509v3 Subject Alternative Name: 
                DNS:localhost, IP Address:127.0.0.1   // here are additional domains / IP addresses the certificate is valid for
            X509v3 Subject Key Identifier: 
                4D:C8:CF:64:87:CE:AC:4F:83:83:E4:6C:AF:1E:89:EC:14:26:3F:D5
            X509v3 Authority Key Identifier: 
                74:C0:E3:B8:BB:49:BA:70:9C:43:09:4A:C1:50:99:7F:21:8C:FB:D7
    Signature Algorithm: sha256WithRSAEncryption
    Signature Value:
        // abbreviated
```